aws/resource_aws_ebs_volume_test.go

```diff
@@ -303,6 +312,38 @@ func TestAccAWSEBSVolume_withTags(t *testing.T) {
 	})
 }
 
+func testAccCheckVolumeDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ec2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ebs_volume" {
+			continue
+		}
+
+		request := &ec2.DescribeVolumesInput{
+			VolumeIds: []*string{aws.String(rs.Primary.ID)},
+		}
+
+		resp, err := conn.DescribeVolumes(request)
+
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidVolume.NotFound" {
```

Comment:
Prefer using `isAWSErr()` instead of working with `awserr` directly:

```suggestion
		if isAWSErr(err, "InvalidVolume.NotFound", "") {
```

Reply:
I fixed.